### DISCUSSION TITLE
Capture output of command ?

### DISCUSSION THREAD

**Discussion by c-michail** [diversenok/NtUtilsLibrary#8] [Q&A] [answered]
Hello,
  How can i capture the output of a command line using NTUtils ?
  
  Thank you

**Comment by diversenok** [comment, marked as answer]
Hi. I just added support for specifying standard input/output handles when using process creation wrappers in https://github.com/diversenok/NtUtilsLibrary/commit/9a8e2b0232f8ca60c580661a9dd5553280f1b965. This includes wrappers for `CreateProcessAsUser`, `CreateProcessWithTokenW`, `CreateProcessWithLogonW`, `NtCreateUserProcess`, `RtlCreateUserProcess`, `RtlCreateUserProcessEx`, and `NtCreateProcessEx`. Keep in mind that the last three don't support inheriting the current console, so they will still spawn a new one (which you might want to `SW_HIDE`), regardless of I/O redirection.

Now it's possible to use anonymous pipes to redirect console I/O, similar to what [this Microsoft sample](https://learn.microsoft.com/en-us/windows/win32/procthread/creating-a-child-process-with-redirected-input-and-output) does.

Here is an example of how to use this functionality to capture output of running a cmd command: https://gist.github.com/diversenok/eb823ea9f81b0f16b9cf8fc97e68c88f

**Reply by c-michail** [reply]
Thank you
